test/Autofac.Test/TypeExtensionsTests.cs

```diff
@@ -170,15 +170,18 @@ private DeclaredConstructorType(bool parameter)
 
         private class DeclaredMethodType
         {
+            [SuppressMessage("CA1822", "CA1822", Justification = "Method needs to be instance for testing.")]
```

Comment:
These test static warnings feel like a candidate for an assembly-level suppression.

Reply:
Fixed with the Test.ruleset.